src/printer.js

```diff
@@ -3837,19 +3828,38 @@ function printJSXElement(path, options, print) {
     return openingLines;
   }
 
+  // Convert `{" "}` to text nodes containing a space.
+  // This makes it easy to turn them into `jsxWhitespace` which
+  // can then print as either a space or `{" "}` when breaking.
+  n.children = n.children.map(child => {
```

Comment:
If possible, would be awesome not to mutate. All sorts of things end up breaking down the line after you do a mutation.